### PR TITLE
Comment `Net.Consensus` related changes

### PR DESCRIPTION
**Pull request by colibrishin**
🌳

### CODE REVIEW

File: Libplanet.Net/Consensus/ConsensusContext.cs
Comment (limebell) [inline Libplanet.Net/Consensus/ConsensusContext.cs:128]:
```suggestion
        /// Starts a consensus for a block of index <paramref name="height/>.
```

File: Libplanet.Net/Consensus/ConsensusContext.cs
Comment (limebell) [inline Libplanet.Net/Consensus/ConsensusContext.cs:133]:
```suggestion
        /// <exception cref="InvalidHeightIncreasingException">Thrown if given height is not same as
```

File: Libplanet.Net/Consensus/ConsensusContext.cs
Comment (limebell) [inline Libplanet.Net/Consensus/ConsensusContext.cs:136]:
```suggestion
        /// <remarks>This method is also called when the tip of the <see cref="BlockChain{T}"/> is
```

File: Libplanet.Net/Consensus/ConsensusContext.cs
Comment (limebell) [inline Libplanet.Net/Consensus/ConsensusContext.cs:181]:
```suggestion
        /// Committing the block to the <see cref="BlockChain{T}"/>.
```
Description is out of context, duplication of <remarks> section.

File: Libplanet.Net/Consensus/ConsensusContext.cs
Comment (limebell) [inline Libplanet.Net/Consensus/ConsensusContext.cs:183]:
```suggestion
        /// <param name="block">A <see cref="Block{T}"/> to committing to the <see cref="BlockChain{T}"/>.
```

File: Libplanet.Net/Consensus/ConsensusContext.cs
Comment (colibrishin) [reply Libplanet.Net/Consensus/ConsensusContext.cs:40]: I was thinking about `BlockChain<T>.ProposeBlock()`. `ConsensusContext` does not use directly but `Context` does, and it seems to be motioned.
https://github.com/planetarium/libplanet/blob/5de07cbc7794e19ee5c1ee3211e198db87de245f/Libplanet/Blockchain/BlockChain.MineBlock.cs#L236-L245
https://github.com/planetarium/libplanet/blob/5de07cbc7794e19ee5c1ee3211e198db87de245f/Libplanet.Net/Consensus/Context.cs#L461-L470

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:941]:
```suggestion
        /// A timeout task for a round if <see cref="ConsensusCommit"/> is received +2/3 any but has
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:892]:
```suggestion
        /// <param name="height">A height that the timeout task is scheduled for.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:894]:
```suggestion
        /// <param name="round">A round that the timeout task is scheduled for.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:919]:
```suggestion
        /// <param name="height">A height that the timeout task is scheduled for.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:921]:
```suggestion
        /// <param name="round">A round that the timeout task is scheduled for.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:946]:
```suggestion
        /// <param name="height">A height that the timeout task is scheduled for.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:948]:
```suggestion
        /// <param name="round">A round that the timeout task is scheduled for.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:867]:
```suggestion
        /// Check whether the round has +2/3 <see cref="ConsensusCommit"/> for the <see cref="Block{T}"/> of hash <paramref name="hash"/>.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:846]:
```suggestion
        /// Check whether the round has +2/3 <see cref="ConsensusVote"/> for the <see cref="Block{T}"/> of hash <paramref name="hash"/>..
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:791]:
```suggestion
        /// Consume every <see cref="ConsensusMessage"/> in the message queue.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:821]:
```suggestion
        /// Get the proposed block and valid round of the given round.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:825]:
```suggestion
        /// <returns>Returns a tuple of proposer and valid round. If proposal for the round does not exist, returns a tuple of
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:826]:
```suggestion
        /// <c>null</c> and <c>null</c>.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:777]:
```suggestion
        /// <param name="step">A new step to set.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:751]:
```suggestion
        /// Creates a <see cref="Vote"/> instance for <see cref="ConsensusVote"/> or
        /// <see cref="ConsensusCommit"/> for the <paramref name="round"/> and the <paramref name="hash"/>.
```

Additionally, how about remove `height` param? It seems only `Height` is used.

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:730]:
```suggestion
        /// <param name="block">A <see cref="Block{T}"/> to validate.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:732]:
```suggestion
        /// <returns>Returns <c>true</c> if block is valid, or otherwise returns <c>false</c>.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:658]:
```suggestion
        /// <param name="round">A round to get proposer.</param>
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:659]:
```suggestion
        /// <returns>Returns designated proposer's <see cref="PublicKey"/> for the <paramref name="round"/>.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:642]:
```suggestion
        /// <returns>A new <see cref="Block{T}"/>.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:391]:
```suggestion
        /// Validate the given <paramref name="message"/> and add it to the message queue.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:367]:
```suggestion
        /// <param name="message">A <see cref="ConsensusMessage"/> that will be handled.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:395]:
```suggestion
        /// <exception cref="InvalidHeightMessageException">Thrown when the height of the message and the context height
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:398]:
```suggestion
        /// <exception cref="InvalidProposerProposeMessageException">Thrown when the <see cref="ConsensusPropose"/>
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:399]:
```suggestion
        /// message has proposer that is not proposer of the current round.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:401]:
```suggestion
        /// <exception cref="InvalidBlockProposeMessageException">Thrown when the <see cref="ConsensusPropose"/>
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:404]:
```suggestion
        /// <exception cref="InvalidValidatorVoteMessageException">Thrown when the signature of the
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:356]:
```suggestion
        /// <param name="round">A round to get the timeout.</param>
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:345]:
```suggestion
        /// <param name="round">A round to get the timeout.</param>
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:343]:
```suggestion
        /// Get the timeout of <see cref="Libplanet.Net.Consensus.Step.PreCommit"/> in the given round.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:354]:
```suggestion
        /// Get the timeout of <see cref="Libplanet.Net.Consensus.Step.Propose"/> in the given round.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:332]:
```suggestion
        /// Get the timeout of <see cref="Libplanet.Net.Consensus.Step.PreVote"/> in the given round.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:334]:
```suggestion
        /// <param name="round">A round to get the timeout.</param>
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:300]:
```suggestion
        /// Add received message to the message queue.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:267]:
```suggestion
        /// Returns a <see cref="Libplanet.Consensus.VoteSet"/> of the given round.
```

File: Libplanet.Net/Consensus/Context.cs
Comment (limebell) [inline Libplanet.Net/Consensus/Context.cs:241]:
```suggestion
        /// <returns>An awaitable task without value. This task
```

**Review by limebell — CHANGES_REQUESTED** [review]
Almost over...

**Review by limebell — APPROVED** [review]
It looks good to me, but please wait for @riemannulus' review before merge.

**Review by riemannulus — APPROVED** [review]
Good Job

**Pull request merged**
